[ALLUXIO-1956] Upgrade Jetty to fix jdk regression. * Used the same v…

nate-cloudability opened the pull request:
…ersion as Spark to reduce the chance of conflicts.

AmplabJenkins commented:
Can one of the admins verify this patch?

apc999 commented:
Jenkins, add to whitelist

apc999 commented:
Thanks for the patch. Could you elaborate a bit what is the error you see before this patch when using Spark (and which version?) ?

tanpinsiang commented:
I have the same problem and found jira ALLUXIO-1956 which point to this pull request.

```
java -version
java version "1.8.0_91"
Java(TM) SE Runtime Environment (build 1.8.0_91-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.91-b14, mixed mode)
```

```
HTTP ERROR 500


Problem accessing /home. Reason:

    Server Error
Caused by:

org.apache.jasper.JasperException: PWC6033: Error in Javac compilation for JSP

PWC6199: Generated servlet error:
The type java.io.ObjectInputStream cannot be resolved. It is indirectly referenced from required .class files


	at org.apache.jasper.compiler.DefaultErrorHandler.javacError(DefaultErrorHandler.java:126)
	at org.apache.jasper.compiler.ErrorDispatcher.javacError(ErrorDispatcher.java:296)
	at org.apache.jasper.compiler.Compiler.generateClass(Compiler.java:372)
	at org.apache.jasper.compiler.Compiler.compile(Compiler.java:433)
	at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:608)
	at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:374)
	at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:476)
	at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:366)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:820)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:652)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:445)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:137)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:574)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:227)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1044)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:372)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:189)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:978)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:135)
	at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:293)
	at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:120)
	at alluxio.web.WebInterfaceGeneralServlet.doGet(WebInterfaceGeneralServlet.java:129)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:820)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:652)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:445)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:137)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:556)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:227)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1044)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:372)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:189)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:978)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:135)
	at org.eclipse.jetty.server.handler.HandlerList.handle(HandlerList.java:52)
	at org.eclipse.jetty.server.handler.HandlerList.handle(HandlerList.java:52)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:116)
	at org.eclipse.jetty.server.Server.handle(Server.java:369)
	at org.eclipse.jetty.server.AbstractHttpConnection.handleRequest(AbstractHttpConnection.java:486)
	at org.eclipse.jetty.server.AbstractHttpConnection.headerComplete(AbstractHttpConnection.java:933)
	at org.eclipse.jetty.server.AbstractHttpConnection$RequestHandler.headerComplete(AbstractHttpConnection.java:995)
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:644)
	at org.eclipse.jetty.http.HttpParser.parseAvailable(HttpParser.java:235)
	at org.eclipse.jetty.server.AsyncHttpConnection.handle(AsyncHttpConnection.java:82)
	at org.eclipse.jetty.io.nio.SelectChannelEndPoint.handle(SelectChannelEndPoint.java:667)
	at org.eclipse.jetty.io.nio.SelectChannelEndPoint$1.run(SelectChannelEndPoint.java:52)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:608)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:543)
	at java.lang.Thread.run(Thread.java:745)
```

nate-cloudability commented:
That is the bug that this patch aims to address. The original ticket raised a concern about using a different jetty version than some other projects (namely Spark). So I updated to the same 9.2.x release they are using.

apc999 commented:
thanks @nate-cloudability for the explanation. I will ask the reporter for ALLUXIO-1956 to test it out. If that works for him too, this patch looks good for me.

apc999 commented:
Jenkins, test this please

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/10360/<h2>Build result: ABORTED</span></h2>[...truncated 4540 lines...]	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)	at java.lang.reflect.Method.invoke(Method.java:606)	at org.codehaus.plexus.classworlds.launcher.Launcher.launchStandard(Launcher.java:330)	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:238)	at jenkins.maven3.agent.Maven31Main.launch(Maven31Main.java:181)	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)	at java.lang.reflect.Method.invoke(Method.java:606)	at hudson.maven.Maven3Builder.call(Maven3Builder.java:136)	at hudson.maven.Maven3Builder.call(Maven3Builder.java:71)	at hudson.remoting.UserRequest.perform(UserRequest.java:120)	at hudson.remoting.UserRequest.perform(UserRequest.java:48)	at hudson.remoting.Request$2.run(Request.java:326)	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)	at java.util.concurrent.FutureTask.run(FutureTask.java:262)	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)	at java.lang.Thread.run(Thread.java:745)[INFO] [INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ alluxio-tests ---[WARNING] JAR will be empty - no content was marked for inclusion!
Test FAILed.

apc999 commented:
Jenkins, test this please

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/10363/<h2>Build result: FAILURE</span></h2>[...truncated 3960 lines...][JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/keyvalue/client/target/alluxio-keyvalue-client-1.2.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-keyvalue-client/1.2.0-SNAPSHOT/alluxio-keyvalue-client-1.2.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/hdfs/pom.xml to org.alluxio/alluxio-underfs-hdfs/1.2.0-SNAPSHOT/alluxio-underfs-hdfs-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/hdfs/target/alluxio-underfs-hdfs-1.2.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-hdfs/1.2.0-SNAPSHOT/alluxio-underfs-hdfs-1.2.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/hdfs/target/alluxio-underfs-hdfs-1.2.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-hdfs/1.2.0-SNAPSHOT/alluxio-underfs-hdfs-1.2.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/swift/pom.xml to org.alluxio/alluxio-underfs-swift/1.2.0-SNAPSHOT/alluxio-underfs-swift-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/swift/target/alluxio-underfs-swift-1.2.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-swift/1.2.0-SNAPSHOT/alluxio-underfs-swift-1.2.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/swift/target/alluxio-underfs-swift-1.2.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-swift/1.2.0-SNAPSHOT/alluxio-underfs-swift-1.2.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/examples/pom.xml to org.alluxio/alluxio-examples/1.2.0-SNAPSHOT/alluxio-examples-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/examples/target/alluxio-examples-1.2.0-SNAPSHOT.jar to org.alluxio/alluxio-examples/1.2.0-SNAPSHOT/alluxio-examples-1.2.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/examples/target/alluxio-examples-1.2.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-examples/1.2.0-SNAPSHOT/alluxio-examples-1.2.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/core/server/pom.xml to org.alluxio/alluxio-core-server/1.2.0-SNAPSHOT/alluxio-core-server-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/core/server/target/alluxio-core-server-1.2.0-SNAPSHOT.jar to org.alluxio/alluxio-core-server/1.2.0-SNAPSHOT/alluxio-core-server-1.2.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/core/server/target/alluxio-core-server-1.2.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-core-server/1.2.0-SNAPSHOT/alluxio-core-server-1.2.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/core/server/target/alluxio-core-server-1.2.0-SNAPSHOT-tests.jar to org.alluxio/alluxio-core-server/1.2.0-SNAPSHOT/alluxio-core-server-1.2.0-SNAPSHOT-tests.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/keyvalue/common/pom.xml to org.alluxio/alluxio-keyvalue-common/1.2.0-SNAPSHOT/alluxio-keyvalue-common-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/keyvalue/common/target/alluxio-keyvalue-common-1.2.0-SNAPSHOT.jar to org.alluxio/alluxio-keyvalue-common/1.2.0-SNAPSHOT/alluxio-keyvalue-common-1.2.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/keyvalue/common/target/alluxio-keyvalue-common-1.2.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-keyvalue-common/1.2.0-SNAPSHOT/alluxio-keyvalue-common-1.2.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/pom.xml to org.alluxio/alluxio-underfs/1.2.0-SNAPSHOT/alluxio-underfs-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/glusterfs/pom.xml to org.alluxio/alluxio-underfs-glusterfs/1.2.0-SNAPSHOT/alluxio-underfs-glusterfs-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/glusterfs/target/alluxio-underfs-glusterfs-1.2.0-SNAPSHOT.jar to org.alluxio/alluxio-underfs-glusterfs/1.2.0-SNAPSHOT/alluxio-underfs-glusterfs-1.2.0-SNAPSHOT.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/underfs/glusterfs/target/alluxio-underfs-glusterfs-1.2.0-SNAPSHOT-sources.jar to org.alluxio/alluxio-underfs-glusterfs/1.2.0-SNAPSHOT/alluxio-underfs-glusterfs-1.2.0-SNAPSHOT-sources.jar[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/integration/pom.xml to org.alluxio/alluxio-integration/1.2.0-SNAPSHOT/alluxio-integration-1.2.0-SNAPSHOT.pom[JENKINS] Archiving /home/jenkins/workspace/Alluxio-Pull-Request-Builder/keyvalue/pom.xml to org.alluxio/alluxio-keyvalue/1.2.0-SNAPSHOT/alluxio-keyvalue-1.2.0-SNAPSHOT.pomchannel stopped
Test FAILed.

apc999 commented:
@nate-cloudability it seems the tests are falling, mostly related to timeout when connecting to webUI server.

nate-cloudability commented:
Thanks @apc999 . Sorry for not running the tests locally first. I was able to reproduce the problem locally and have fixed it with the latest comment. 

Looking into the issue it appears that previously this code used a port of 0 which would cause the jdk to pick a random available port. That's why it was starting the server first then setting the configuration of which port it actually bound to.

I fixed the issue by assigning the socket connector as a member variable for later reference. I also updated getLocalPort() and getBindHost() methods to be reflective the actual host and port that were bound to.

yupeng9 commented:
Jenkins, test this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/10385/
Test PASSed.

yupeng9 commented on core/server/pom.xml:
```diff
@@ -122,6 +122,12 @@
     </dependency>
+    <dependency>
+      <groupId>org.eclipse.jetty.toolchain</groupId>
+      <artifactId>jetty-jsp-jdt</artifactId>
```
Can you explain what this dependency is for?

nate-cloudability replied:
Without that dependency the jsps fail to compile with the following exception.

`org.apache.jasper.JasperException: PWC6349: Cannot find a java compiler for compilation.  If running with JDK 5 or before, Ant or JDT compiler can be used, if the corresponding jars and bridge classes (org.apache.jasper.compiler.AntJavaCompiler or org.apache.jasper.compiler.JDTJavaCompiler) are included
	at org.apache.jasper.compiler.DefaultErrorHandler.jspError(DefaultErrorHandler.java:92)
	at org.apache.jasper.compiler.ErrorDispatcher.dispatch(ErrorDispatcher.java:378)
	at org.apache.jasper.compiler.ErrorDispatcher.jspError(ErrorDispatcher.java:119)
	at org.apache.jasper.compiler.Compiler.initJavaCompiler(Compiler.java:773)
	at org.apache.jasper.compiler.Compiler.<init>(Compiler.java:140)
	at org.apache.jasper.JspCompilationContext.createCompiler(JspCompilationContext.java:288)
	at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:622)
	at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:375)
	at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:473)
	at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:377)
	at org.eclipse.jetty.jsp.JettyJspServlet.service(JettyJspServlet.java:103)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:812)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:587)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:595)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:191)
	at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:72)
	at alluxio.web.WebInterfaceBrowseServlet.doGet(WebInterfaceBrowseServlet.java:287)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:687)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:812)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:587)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:577)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerList.handle(HandlerList.java:52)
	at org.eclipse.jetty.server.handler.HandlerList.handle(HandlerList.java:52)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
	at org.eclipse.jetty.server.Server.handle(Server.java:499)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:311)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:257)
	at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:544)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555)
	at java.lang.Thread.run(Thread.java:745)
`

yupeng9 replied:
gotcha. Thanks.

yupeng9 replied:
@nate-cloudability I found this dependency is under GPL license. Alluxio is under the Apache 2.0 license, and the [Apache Software Foundation's official guideline](http://www.apache.org/legal/resolved.html#category-b) is not to include LGPL licenses. Could you find an alternative to JDT compiler? Thanks.

nate-cloudability replied:
Pushed a change that uses the Apache licensed JSP libs.

yupeng9 commented:
Thanks for the contribution. Could you sign the contribution license agreement?

nate-cloudability commented:
@yupeng9 The license agreement is waiting with my company's legal department. Nobody foresees any issue so I should be able to return it soon.

yupeng9 commented:
LGTM

yupeng9 commented:
@nate-cloudability this PR is good to merge after signing cla.

yupeng9 commented:
jenkins, test this please

yupeng9 commented on core/server/src/main/java/alluxio/web/UIWebServer.java:
```diff
@@ -67,22 +72,38 @@ public UIWebServer(ServiceType service, InetSocketAddress address, Configuration
     mWebAppContext.setContextPath(AlluxioURI.SEPARATOR);
     File warPath = new File(mConfiguration.get(Constants.WEB_RESOURCES));
     mWebAppContext.setWar(warPath.getAbsolutePath());
+
```
Could you explain why these configuration and attribute settings are needed?

nate-cloudability replied:
These were taking from the examples provided by [Jetty](https://www.eclipse.org/jetty/documentation/9.3.x/embedded-examples.html#embedded-webapp-jsp)

I omitted the comments. But they are 

>  // This webapp will use jsps and jstl. We need to enable the
        // AnnotationConfiguration in order to correctly
        // set up the jsp container

and 

>  // Set the ContainerIncludeJarPattern so that jetty examines these
        // container-path jars for tlds, web-fragments etc.
        // If you omit the jar that contains the jstl .tlds, the jsp engine will
        // scan for them instead.

yupeng9 replied:
Thanks. Could you carry over the comments? So that we can better understand why they are needed.

nate-cloudability replied:
Sure. Pushed change with comments.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/10689/<h2>Build result: ABORTED</span></h2>[...truncated 4269 lines...]	at org.jvnet.hudson.maven3.launcher.Maven31Launcher.main(Maven31Launcher.java:132)	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)	at java.lang.reflect.Method.invoke(Method.java:606)	at org.codehaus.plexus.classworlds.launcher.Launcher.launchStandard(Launcher.java:330)	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:238)	at jenkins.maven3.agent.Maven31Main.launch(Maven31Main.java:181)	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)	at java.lang.reflect.Method.invoke(Method.java:606)	at hudson.maven.Maven3Builder.call(Maven3Builder.java:136)	at hudson.maven.Maven3Builder.call(Maven3Builder.java:71)	at hudson.remoting.UserRequest.perform(UserRequest.java:120)	at hudson.remoting.UserRequest.perform(UserRequest.java:48)	at hudson.remoting.Request$2.run(Request.java:326)	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)	at java.util.concurrent.FutureTask.run(FutureTask.java:262)	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)	at java.lang.Thread.run(Thread.java:745)[INFO] [INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ alluxio-tests ---
Test FAILed.

yupeng9 commented:
jenkins, test this please

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/10692/
Test FAILed.

nate-cloudability commented:
Looks like the CI server is having trouble checking out the code. Did I do something wrong with the merge?

gpang commented:
Jenkins, test this please

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/10694/<h2>Build result: ABORTED</span></h2>[...truncated 4156 lines...]	at org.jvnet.hudson.maven3.launcher.Maven31Launcher.main(Maven31Launcher.java:132)	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)	at java.lang.reflect.Method.invoke(Method.java:606)	at org.codehaus.plexus.classworlds.launcher.Launcher.launchStandard(Launcher.java:330)	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:238)	at jenkins.maven3.agent.Maven31Main.launch(Maven31Main.java:181)	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)	at java.lang.reflect.Method.invoke(Method.java:606)	at hudson.maven.Maven3Builder.call(Maven3Builder.java:136)	at hudson.maven.Maven3Builder.call(Maven3Builder.java:71)	at hudson.remoting.UserRequest.perform(UserRequest.java:120)	at hudson.remoting.UserRequest.perform(UserRequest.java:48)	at hudson.remoting.Request$2.run(Request.java:326)	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)	at java.util.concurrent.FutureTask.run(FutureTask.java:262)	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)	at java.lang.Thread.run(Thread.java:745)[INFO] [INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ alluxio-tests ---
Test FAILed.

yupeng9 commented:
Jenkins, test this please

yupeng9 commented:
Thanks for addressing the comments. From the build failure `java.lang.RuntimeException: Failed to start cluster. Timed out waiting for master to serve web
	at alluxio.master.AbstractLocalAlluxioCluster.waitAndCheckTimeout(AbstractLocalAlluxioCluster.java:167)`, it seems to be caused by the change. Could you run the tests locally see if you can reproduce it?

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/10700/<h2>Build result: ABORTED</span></h2>[...truncated 4162 lines...]	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)	at java.lang.reflect.Method.invoke(Method.java:606)	at org.codehaus.plexus.classworlds.launcher.Launcher.launchStandard(Launcher.java:330)	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:238)	at jenkins.maven3.agent.Maven31Main.launch(Maven31Main.java:181)	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)	at java.lang.reflect.Method.invoke(Method.java:606)	at hudson.maven.Maven3Builder.call(Maven3Builder.java:136)	at hudson.maven.Maven3Builder.call(Maven3Builder.java:71)	at hudson.remoting.UserRequest.perform(UserRequest.java:120)	at hudson.remoting.UserRequest.perform(UserRequest.java:48)	at hudson.remoting.Request$2.run(Request.java:326)	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)	at java.util.concurrent.FutureTask.run(FutureTask.java:262)	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)	at java.lang.Thread.run(Thread.java:745)[INFO] [INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ alluxio-tests ---[WARNING] JAR will be empty - no content was marked for inclusion![INFO] Building jar: /home/jenkins/workspace/Alluxio-Pull-Request-Builder/tests/target/alluxio-tests-1.2.0-SNAPSHOT.jar
Test FAILed.

nate-cloudability commented:
@yupeng9 I'm still looking into the test failure. The exception leads me to believe there is some classpath issue. [Exception](https://gist.github.com/nate-cloudability/eb6aa39317f59c881a91a896e6a9fae1)

The test passes if run in Intellij. I'll keep trying to fix as time allows.

yupeng9 commented:
Thanks for investigating the issue. Reading the dependency graph via maven might be helpful.

nate-cloudability commented:
Alright sorry for the delay @yupeng9 . I have fixed the issue locally by removing the XML and Annotation classes. All tests pass for me and Alluxio runs fine locally.

yupeng9 commented:
Jenkins, test this please

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Alluxio-Pull-Request-Builder/10864/
Test PASSed.

yupeng9 commented:
LGTM. Thanks for the fix!
